### DISCUSSION TITLE
Webhook for Payment Request? Or Invoice event containing reference to the Payment Request ID?

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2643] [Ideas / Feature Requests]
We currently have 6 webhooks for invoices, but none for Payment Requests and I want to know when a Payment Request has settled.

Since Payment Requests also create invoices, we should be able to manage.

At present, it seams that an invoice has an order ID that starts with "PAY_REQUEST_" if it was created for a PR.

Is this the best way to link these 2? I.e. I have to look for this prefix and strip it in order to be able to load the Payment Request?

A separate field or separate event may be cleaner.

Just looking for best practise atm. Not looking to refactor as Payment Requests are on the roadmap to be moved to a plugin.

**Comment by Kukks** [comment]
concept ACK

**Reply by woutersamaey** [reply]
Okay so I am on to something here? What would you suggest?

We can do all kinds of things from easy to refactoring:
1. Add an extra field `PaymentRequestId` to the webhook payload?
2. Add an extra field `PaymentRequestId` to the `Invoice.Metadata`?
3. Create separate webhooks for Payment Requests (not what I would pick, but naming it for completeness sake)

I'd suggest to do 1 and 2.

**Reply by woutersamaey** [reply]
Already solved number 2 in #2644

**Reply by woutersamaey** [reply]
Also fixed number 1 + event put the Payment Request ID on the invoice detail page with a link, so it is visible for people who don't talk API.
